ucp/_libs/ucx_address.pyx

```diff
@@ -41,10 +63,7 @@ cdef class UCXAddress:
         cdef size_t length
         status = ucp_worker_get_address(ucp_worker, &address, &length)
         assert_ucs_status(status)
-        try:
-            return UCXAddress(int(<uintptr_t>address), length)
-        finally:
-            ucp_worker_release_address(ucp_worker, address)
+        return UCXAddress(int(<uintptr_t>address), length, worker=worker)
```

Comment:
Do we still need to convert to a Python `int` given the constructor accepts `uintptr_t`?

```suggestion
        return UCXAddress(<uintptr_t>address, length, worker=worker)
```